src/KrakenClient.cs

```diff
@@ -73,18 +64,13 @@ public partial class KrakenClient : IDisposable
         public KrakenClient(
             string apiKey,
             string privateKey,
-            RateLimit rateLimit = RateLimit.None)
+            RateLimitInfo rateLimit = null)
         {
             ApiKey = apiKey ?? throw new ArgumentNullException(nameof(apiKey));
             PrivateKey = privateKey ?? throw new ArgumentNullException(nameof(privateKey));
 
-            RateLimit = rateLimit;
-            if (TierInfo.TryGetValue(rateLimit, out var info))
-            {
-                _privateApiRateLimiter = new RateLimiter(info.Limit, info.DecreaseTime, new Stopwatch());
-                // If private rate limiter enabled, also enable public rate limiter.
-                _publicApiRateLimiter = new RateLimiter(20, TimeSpan.FromSeconds(1), new Stopwatch());
-            }
+            RateLimit = rateLimit ?? RateLimitInfo.Tier2; //Unless the user makes a choice default to limit.
```

Comment:
Honestly, I don't get this one here. Why would the RateLimit be anything but None by default?

Reply:
@wklm from what I understood None means no limit. This means a new user will be throttled(rejected service for 15 minutes) after only 15 requests in a quick sequence.  The default should be to not have the user shoot themselves in the foot.

Since this library interfaces with a specific service with clearly defined limits I don't see the use for an unlimited library.